GVRf/Extensions/gvrf-physics/src/main/java/org/gearvrf/physics/GVRWorld.java

```diff
@@ -113,9 +133,15 @@ public void removeBody(GVRRigidBody gvrBody) {
 
     @Override
     public void onDrawFrame(float frameTime) {
-        NativePhysics3DWorld.step(getNative(), frameTime);
-
+        mFrameTime += frameTime;
+        if (mIsProcessing) {
```

Comment:
@NolaDonato Why do you think onDrawFrame needs to make provisions to be reentrant? It is only called once per frame by the view manager. While technically it is a public method and the user could call it any time, they shouldn't and we mention that somewhere. And the method doesn't have to be public at all to do what it does.